### PR TITLE
addpkg: wakatime

### PR DESCRIPTION
**Pull request by piggynl**

### CODE REVIEW

File: wakatime/riscv64.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ checkdepends=('svn')
+-source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}")
+-sha256sums=('SKIP')
++source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}"
++        "main_test-arch.patch::https://github.com/wakatime/wakatime-cli/pull/642.patch")
++sha256sums=('SKIP'
++            'eeea51959573aba7091357fa428fb60a90cc4148b788cc5d8d12f4adb4753461')
+ 
+-_binname="${pkgname}-cli-linux-amd64"
++_binname="${pkgname}-cli-linux-${CARCH}"
```
Comment (Xeonacid) [inline wakatime/riscv64.patch:15]: This won't work for x86_64, so I think directly patching it to `riscv64` may be more clear.

**Comment by cubercsl** [comment]
Need to be useful after https://github.com/wakatime/wakatime-cli/pull/642 is merged. Otherwise the test will fail.

**Review by r-value — CHANGES_REQUESTED** [review]
Can't reproduce. Build fails in `check()`.
Error log:
```plain
==> Starting check()...
go: downloading github.com/stretchr/testify v1.7.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
./build/wakatime-cli-linux-riscv64 --log-file /tmp/TestSendHeartbeats2731894363/002/1416661128 --api-url http://127.0.0.1:42365 --key 00000000-0000-4000-8000-000000000000 --config /tmp/TestSendHeartbeats2731894363/001/wakatime.cfg2941296879 --entity testdata/main.go --cursorpos 12 --offline-queue-file /tmp/TestSendHeartbeats2731894363/001/3667903834 --lineno 42 --lines-in-file 100 --time 1585598059 --hide-branch-names .* --project wakatime-cli --write --verbose


failed to run command ./build/wakatime-cli-linux-riscv64 --log-file /tmp/TestSendHeartbeats2731894363/002/1416661128 --api-url http://127.0.0.1:42365 --key 00000000-0000-4000-8000-000000000000 --config /tmp/TestSendHeartbeats2731894363/001/wakatime.cfg2941296879 --entity testdata/main.go --cursorpos 12 --offline-queue-file /tmp/TestSendHeartbeats2731894363/001/3667903834 --lineno 42 --lines-in-file 100 --time 1585598059 --hide-branch-names .* --project wakatime-cli --write --verbose: fork/exec ./build/wakatime-cli-linux-riscv64: no such file or directory
FAIL    command-line-arguments  0.085s
FAIL
==> ERROR: A failure occurred in check().
    Aborting...
```

File: wakatime/riscv64.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+-source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}")
+-sha256sums=('SKIP')
++source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}"
++        "main_test-arch.patch::https://github.com/wakatime/wakatime-cli/pull/645.patch")
++sha256sums=('SKIP'
++            'a53e0d001d2a8e0bcf1e7c0ceb82a82d0ece4b65663af56387d898fb5b2e1e83')
+ 
+-_binname="${pkgname}-cli-linux-amd64"
++_GOARCH=$(go env GOARCH)
++_binname="${pkgname}-cli-linux-${_GOARCH}"
```
Comment (piggynl) [inline wakatime/riscv64.patch:16]: Hi, it appears that `go` has not been installed when this line is executed, which results in `_GOARCH` being an empty string. Do you have any suggestions for a solution?

Comment (cubercsl) [reply wakatime/riscv64.patch:16]: Can we place these lines in `build()`?

Comment (piggynl) [reply wakatime/riscv64.patch]: Thanks!

Comment (r-value) [reply wakatime/riscv64.patch]: Actually I think it might be better to initialize the enviroment variable in `prepare()`?

Comment (r-value) [reply wakatime/riscv64.patch]: Actually I think it might be better to initialize the enviroment variable in `prepare()`?

Comment (piggynl) [reply wakatime/riscv64.patch]: Hi @r-value, by Arch Linux wiki page [Creating packages](https://wiki.archlinux.org/title/Creating_packages#prepare()), `prepare()` is used to prepare sources and will not be run if `--noextract` is specified for `makepkg`. However, in this case, the variable `_binname` should always be set regardless of whether the source code is extracted.

> **prepare()**
>
> With this function, commands that are used to prepare sources for building are run, such as [patching](https://wiki.archlinux.org/title/Patching_in_ABS). This function runs right after package extraction, before [pkgver()](https://wiki.archlinux.org/title/Creating_packages#pkgver()) and the build function. If extraction is skipped (makepkg --noextract), then prepare() is not run.

Comment (r-value) [reply wakatime/riscv64.patch]: > Hi @r-value, by Arch Linux wiki page [Creating packages](https://wiki.archlinux.org/title/Creating_packages#prepare()), `prepare()` is used to prepare sources and will not be run if `--noextract` is specified for `makepkg`. However, in this case, the variable `_binname` should always be set regardless of whether the source code is extracted.
> 
> > **prepare()**
> > With this function, commands that are used to prepare sources for building are run, such as [patching](https://wiki.archlinux.org/title/Patching_in_ABS). This function runs right after package extraction, before [pkgver()](https://wiki.archlinux.org/title/Creating_packages#pkgver()) and the build function. If extraction is skipped (makepkg --noextract), then prepare() is not run.

True enough.

**Review by r-value — APPROVED** [review]

File: wakatime/riscv64.patch [hunk 3]
```diff
@@ -0,0 +1,36 @@
+--- PKGBUILD
++++ PKGBUILD
+@@ -14,12 +14,21 @@ license=('BSD')
+ depends=('glibc')
+ makedepends=('git' 'go')
+ checkdepends=('svn')
+-source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}")
+-sha256sums=('SKIP')
++source=("${pkgname}-cli::git+https://github.com/${pkgname}/${pkgname}-cli#tag=v${pkgver}"
++        "main_test-arch.patch::https://github.com/wakatime/wakatime-cli/pull/645.patch")
```
Comment (felixonmars) [inline wakatime/riscv64.patch:10]: The patch name looks general enough that may conflict with others when `SRCDEST` is set. Please prefix it with something like `$pkgname-`.

**Review by felixonmars — CHANGES_REQUESTED** [review]

**Pull request merged**